sdk/identity/azure-identity/azure/identity/_credentials/user_password.py

```diff
@@ -59,6 +60,12 @@ class UsernamePasswordCredential(InteractiveCredential):
     """
 
     def __init__(self, client_id: str, username: str, password: str, **kwargs: Any) -> None:
+        warnings.warn(
```

Comment:
Is this `@deprecated` decorator something that we could use? https://docs.python.org/3/library/warnings.html#warnings.deprecated

It would be nice to have something like this that the Learn API docs can see to classify something as deprecated.

Reply:
Unfortunately, we can't. That decorator was just added in Python 3.13, while we still need to maintain support for 3.8+. Too hot off the press.

And I agree, having that deprecated indicator in the Learn docs. Will need to investigate to see the process behind that.

Reply:
Makes sense. I have a thread going with the Learn platform PM about the indicators.